pymatgen/transformations/advanced_transformations.py

```diff
@@ -1360,3 +1361,213 @@ def inverse(self):
+                                np.linalg.norm(proposed_sc_lat_vecs[1]),
+                                np.linalg.norm(proposed_sc_lat_vecs[2])]
+                maxNorm = max(scBasesNorms)
+                maxIndex = scBasesNorms.index(maxNorm)
+                idx = list(range(3))
+                idx.remove(maxIndex)
+                a = proposed_sc_lat_vecs[maxIndex]
+                b = proposed_sc_lat_vecs[idx[0]]
+                c = proposed_sc_lat_vecs[idx[1]]
+                projb_a = self.proj(b, a)
+                projc_a = self.proj(c, a)
+
+                if np.linalg.norm(projb_a) > np.linalg.norm(projc_a):
+                    length = np.linalg.norm(a - projb_a)
+                else:
+                    length = np.linalg.norm(a - projc_a)
+                width = math.sqrt(np.linalg.norm(b) ** 2 - np.linalg.norm(projb_a) ** 2)
+                ab_normal = np.cross(a, b)  # get normal direction from AB plane
+                height = np.linalg.norm(self.proj(c, ab_normal))  # project c onto AB plane normal
+
+                self.smallest_dim = min([length, width, height])
+
+                # Get number of atoms
+                superstructure = SupercellTransformation(trans_mat).apply_transformation(self.structure)
+                num_at = superstructure.num_sites
+
+                if self.smallest_dim >= hard_sc_size_threshold \
+                        and num_at >= self.min_atoms and num_at <= self.max_atoms:
+                    return superstructure, trans_mat
+                else:
+                    # Increase threshold until proposed supercell meets requirements
+                    target_sc_size += 0.1
+                    if num_at > self.max_atoms:
+                        raise AttributeError('While trying to solve for the supercell, the max'
+                                             'number of atoms was exceeded. Try lowering the number'
+                                             'of nearest neighbor distances.')
+
+class PerturbedSupercellsTransformation:
+    """
+        Uses a supercell Structure, a transformation matrix, a min/max displacement,
+        number of displacement values, and number of supercells per displacement
+        to generate a list of randomly perturbed supercells from a given supercell.
+
+        Args:
+            original_supercell (Structure): supercell to displace, typically already relaxed
+            max_displacement_val (float): maximum displacement value for perturbing the supercell (in Angstroms)
+            min_displacement_val (float): minimum displacement value for perturbing the supercell (in Angstroms)
+            num_displacements (int): number of unique displacement values to try, bounded by
+                min_displacement and max_displacement. This argument is ignored if min_displacement
+                is not passed as an argument.
+            scs_per_displacement_val (int): number of perturbed supercells to generate for each unique
```

Comment:
Rename to `num_structures`.

Reply:
Renaming 'num_displacements' to 'num_structures' might be a little misleading since the number of structures is actually 'num_displacements' times 'scs_per_displacement_val'. Would 'num_unique_displacement_vals' be more descriptive?